codebuild/bin/s2n_open_fds_test.py

```diff
@@ -0,0 +1,81 @@
+import os
+import re
+import sys
+
+# Allow stdin, stdout, stderr, and the file descriptor for dynamic analysis log to remain open at exit
+ACCEPTABLE_OPEN_FDS = 4
+ERROR_EXIT_CODE = 123
+
+analysis_file_location = "../../build/Testing/Temporary"
+analysis_file_pattern = re.compile(r"^LastDynamicAnalysis.*")
+# This regular expression captures valgrind 3.13 and valgrind 3.18+ log
+fd_pattern = re.compile(r"FILE DESCRIPTORS: \d+ open(?: \(\d+ std\))? at exit.$")
+error_message_start_pattern = re.compile(r"^Running .*/s2n_.*_test.c.*")
+error_message_end_pattern = re.compile(r"^<end of output>$")
+
+
+def open_analysis_file(path):
+    for file_name in os.listdir(path):
+        if analysis_file_pattern.match(file_name):
+            file = open(os.path.join(path, file_name), 'r')
+    return file
+
+
+def print_error_message(file_name, error_message):
+    # Take the path to the failing test file
+    print(f"{file_name.split(' ')[1]} leaks file descriptor(s):\n")
+    print(''.join(error_message))
+    return 0
+
+
+def read_analysis_file(file):
```

Comment:
I wonder if we could simplify this process?

Instead of parsing/using regexs, we could do a substring match against `FILE DESCRIPTORS: `. And then just print out the surronding lines if one of the `FILE DESCRIPTORS` was more than `ACCEPTABLE_OPEN_FDS`.

Reply:
> I wonder if we could simplify this process?

I have changed the testing logic (see PR descriptions for further details). I believe the new check will simplify this process.

> Instead of parsing/using regexs, we could do a substring match against FILE DESCRIPTORS: . And then just print out the surronding lines if one of the FILE DESCRIPTORS was more than ACCEPTABLE_OPEN_FDS.

The original logic for checking if there are four open fds is incorrect. There are some tests that intentionally close one std. If a test close a std, and leak another fd, that is still problematic, but the previous logic (checking if there are four open fds) will treat it as correct.